iceoryx_utils/include/iceoryx_utils/internal/concurrent/sofi.inl

```diff
@@ -116,6 +116,7 @@ inline bool SoFi<ValueType, CapacityValue>::popIf(ValueType& valueOut, const Ver
             ///        changed
             if (m_readPosition.load(std::memory_order_relaxed) == currentReadPosition && verificator(valueOut) == false)
             {
+                nextReadPosition = currentReadPosition;
```

Comment:
A bugfix should always result in a new unit test case. Can you add one that reconstructs the bug?

Reply:
Nice idea but sadly this is impossible since it is a freaky race condition. With total control over all threads it would be possible but there is no such thing available.

Reply:
@GerdHirsch always created check-points in the code. I guess we would need something similar here. However, you are right, the rule "bugfix==at least one additional test case" is a general rule. I've not fully understood the race yet, can we somehow formalize it?

Reply:
I think that the check point approach is not the right solution for this particular problem.
@mossmaurice The race is explained in detail in the following issue: https://github.com/eclipse/iceoryx/issues/182